### ISSUE TITLE
HTTP::Server[.new, .listen] has changed in crystal 0.25

### ISSUE THREAD

**Issue by Timbus** [obsidian/orion#7]
The example code in this repo currently throws this error at compile time:
```
Router.listen(port: 3000)
              ^~~~~~

in lib/orion/src/orion/router.cr:35: 'HTTP::Server.new' is expected to be invoked with a block, but no block was given

    server = HTTP::Server.new(host: host, port: port, handler: router)
```
In 0.25 you pass the host and port number to HTTP::Server.listen, or HTTP::Server.tcp_bind.

I think you need to replace Orion::Router.listen with something like: 
```crystal
  def self.listen(*, autoclose = true, host = "127.0.0.1", port = 3000, reuse_port = false)
    router = new(autoclose: autoclose)
    server = HTTP::Server.new(handler: router)
    server.listen(host: host, port: port, reuse_port: reuse_port)
  end
```

**Comment by Timbus** [comment]
PS: Absolutely excellent library. Keeps my APIs really clean and testable!

**Comment by jwaldrip** [comment]
@Timbus, working on a fix right now.

**Comment by jwaldrip** [comment]
@Timbus fixed in orion `v1.5.0`
